storage: expose site of error return

tbg opened the pull request:
Please ignore the first commit, that's #18680.

----

Hopefully this can shed some light on #18084. A similar patch has already
landed in 1.0.6.

I tested this manually by returning an error from `handleRaftReady`:

> store.go:3580: %s: %s | myexplanation; <*errors.fundamental>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18681)
<!-- Reviewable:end -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 7 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18681#-:-Kua6IT-BzZEZhxXWkKE:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: expose site of error return

Hopefully this can shed some light on #18084.
A similar patch has already landed in 1.0.6.

I tested this manually by returning an error from `handleRaftReady`:

> store.go:3580: %s: %s | myexplanation; <*errors.fundamental>
